### PR TITLE
Raise exception in read_csv when prefix is set, but not used because a header exists

### LINKED ISSUE #27394: pd.read_csv prefix parameter seems do not works
#### Code Sample, a copy-pastable example if possible
```test.csv
field,data,values
text1,34,hh,fail
text2,76,tt,fail2
```
```python
import pandas as pd

if __name__ == "__main__":
    data_list = pd.read_csv('test.csv', prefix="X")
    print(data_list)

```
#### Problem description
A documentation said: 
```
prefix : str, optional
    Prefix to add to column numbers when no header, e.g. ‘X’ for X0, X1, …
```
If I understood correctly this documentation I should get this

#### Expected Output
```
field    data    values  X0
text1     34    hh        fail
text2     76    tt         fail2
```
But I got this anyway:

#### Current Output
```
          field  data    values
text1     34    hh      fail
text2     76    tt        fail2
```
#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
commit: None
python: 3.6.3.final.0
python-bits: 64
OS: Windows
OS-release: 8.1
machine: AMD64
processor: Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.24.2
pytest: None
pip: 19.1.1
setuptools: 40.1.0
Cython: None
numpy: 1.15.0
scipy: None
pyarrow: None
xarray: None
IPython: None
sphinx: None
patsy: None
dateutil: 2.7.5
pytz: 2018.5
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 2.2.3
openpyxl: 2.5.5
xlrd: None
xlwt: None
xlsxwriter: None
lxml.etree: 4.1.1
bs4: 4.6.0
html5lib: None
sqlalchemy: 1.2.19
pymysql: None
psycopg2: None
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None
None

</details>

### PR DESCRIPTION
**Pull request by PraneethKhanna**
Raised an error if argument prefix is set when there are headers present during pandas read_csv.

- [x] closes #27394
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by mroeschke — CHANGES_REQUESTED** [review]
Needs test and whatsnew entry in 1.0.0

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -1930,6 +1930,9 @@ def __init__(self, src, **kwds):
                 ]
             else:
                 self.names = list(range(self._reader.table_width))
+        else:
+            if self.prefix:
```
Comment (datapythonista) [inline pandas/io/parsers.py]: probably worth using `elif`

File: pandas/io/parsers.py [hunk 2]
```diff
@@ -1930,6 +1930,9 @@ def __init__(self, src, **kwds):
...
+                raise ValueError("argument prefix must be None if header not None")
```
Comment (datapythonista) [inline pandas/io/parsers.py]: The error message is not 100% correct, since the header can be provided with `names`.

**Review by datapythonista — COMMENTED** [review]
This change can break code, not sure if we should start by a warning and break later.

File: pandas/io/parsers.py (hunk 1 above)
Comment (PraneethKhanna) [reply pandas/io/parsers.py]: yes, will change that

**Comment by datapythonista** [comment]
@PraneethKhanna do you have time to add the requested test and the note in the release notes (aka whatsnew)?

**Comment by WillAyd** [comment]
Looks like this is stale but @PraneethKhanna if you'd like to pick back up please ping!

**Pull request closed without merging**